orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -344,19 +614,22 @@ Status TrainingRunner::TrainingLoop(IDataLoader& training_data_loader, IDataLoad
         }
 
         // Print some info when reaching the end of the batch.
-        printf("Round %d, Step: %d, epoch: %d, batch: %d/%d, shard_iteration: %d/%d, time: %.2f ms, throughput: %.2f ex/sec \n",
-               static_cast<int>(round_),
-               static_cast<int>(step_),
-               static_cast<int>(epoch),
-               static_cast<int>(batch),
-               static_cast<int>(batch_num_cur_shard),
-               static_cast<int>(shard_it + 1),
-               static_cast<int>(num_shards_to_visit),
-               duration_seconds.count() * 1000,
-               params_.batch_size * (step_ - step_start) / total_time);
-        printf("Training data range: [%d - %d)\n",
-               static_cast<int>(batch * params_.batch_size),
-               static_cast<int>((batch + 1) * params_.batch_size - 1));
+        // When pipeline is enabled, the first stage, idexed by 0, computes the start and end of a batch.
+        if (params_.num_pipeline_stages == 1 || pipeline_context_.pipeline_stage_id == 0) {
```

Comment:
>pipeline_context_.pipeline_stage_id == 0 [](start = 48, length = 40)

would be nice to see output from different pipeline stages too, if needed. MPI can choose to output only rank-0 as we did for Data parallelism.

Reply:
Ok. This condition is removed and all ranks may print the following content.